### PR TITLE
[Translation] [Portuguese Brazil]

### PR DESCRIPTION
**Pull request by luciormoraes**
This PR includes the following files:

- slices.article
- generics-underlying-types.article
- generics-struct-types.article

### CODE REVIEW

File: _content/tour/por/slices.article [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
-- *Example* *9:* Efficient Traversals
-- *Example* *10:* Three index slicing
+- *Exemplo* *1:* Declarar e Obter o comprimento
+- *Exemplo* *2:* Tipos por Referência
+- *Exemplo* *3:* Anexando slices
+- *Exemplo* *4:* Obter slices de slices
+- *Exemplo* *5:* Slices e Referências
+- *Exemplo* *6:* Strings e slices
+- *Exemplo* *7:* Funções Variádicas
+- *Exemplo* *8:* Range Mecânica do Range
```
Comment (rschio) [inline _content/tour/por/slices.article:26]:
```suggestion
- *Exemplo* *8:* Mecânica do Range
```

File: _content/tour/por/slices.article [hunk 2]
```diff
@@ -36,9 +38,9 @@ very beneficial for all Go programmers to learn how slices work, and how to use
 .play slices/example9.go 
 .play slices/example10.go
 
-** Slice Construction
+** Construção de Slice
 
-Constructing a slice can be done in several ways.
+A construção de uma fatia pode ser feita de várias maneiras.
```
Comment (rschio) [inline _content/tour/por/slices.article:43]:
```suggestion
A construção de uma slice pode ser feita de várias maneiras.
```

File: _content/tour/por/slices.article [hunk 3]
```diff
@@ -1,29 +1,31 @@
 Slices
-Slices are an incredibly important data structure in Go. They form the basis for how we manage and manipulate data in a flexible, performant, and dynamic way. It's very beneficial for all Go programmers to learn how slices work, and how to use them.
+`Slices` são uma estrutura de dados incrivelmente importante em Go. Elas formam a base de como gerenciamos e
+manipulamos dados de forma flexível, eficiente e dinâmica. É muito benéfico para todos os programadores
+em Go aprenderem como `Slices` funcionam e como usá-los.
```
Comment (rschio) [inline _content/tour/por/slices.article:4]: The section description needs to be in one single line or the program returns a parse error.

```suggestion
`Slices` são uma estrutura de dados incrivelmente importante em Go. Elas formam a base de como gerenciamos e manipulamos dados de forma flexível, eficiente e dinâmica. É muito benéfico para todos os programadores em Go aprenderem como `Slices` funcionam e como usá-los.
```

File: _content/tour/por/slices.article [hunk 4]
```diff
@@ -102,17 +102,20 @@ to move the data around my program. This includes declaring fields on a type.
         Z []bool
     }
 
-This goes for all of Go’s internal data structures (slices, maps, channels, interfaces, and functions).
+Isso se aplica a todas as estruturas de dados internas do Go (slices, maps, channels, interfaces e functions).
 
 One reason to switch to pointer semantics is if you need to share the slice for a
 decoding or unmarshaling operation. Using pointers for these types of operations
 are ok, but document this if it’s not obvious.
```
Comment (rschio) [inline _content/tour/por/slices.article:109]:
```suggestion
```

File: _content/tour/por/slices.article [hunk 5]
```diff
@@ -102,17 +102,20 @@ to move the data around my program. This includes declaring fields on a type.
...
+Uma razão para mudar para a semântica de ponteiro é se você precisar compartilhar o `slice` para uma operação
+de decodificação ou desmarcação. Usar ponteiros para esses tipos de operações é aceitável, mas documente
+isso se não for óbvio.
```
Comment (rschio) [inline _content/tour/por/slices.article:112]: "decodificação ou desmarcação" -> "decoding ou unmarshal" or "decodificação e unmarshal", desmarcação is unusual in my opinion.

File: _content/tour/por/slices.article [hunk 6]
```diff
@@ -141,83 +144,77 @@ Output:
...
-are constructed by doubling the size of the existing array. Once the backing array
-grows past 1024 elements, growth happens at 25%. 
+Quando o `array` subjacente tem 1024 elementos de capacidade ou menos, novos `arrays` subjacentes são construídos
+dobrando o tamanho do `array` existente. Uma vez que o `array` subjacente cresce além de 1024 elementos,
+o crescimento ocorre em 25%.
 
-*NOTE:* *How* *the* *append* *function* *grows* *the* *capacity* *of* *the*
-*backing* *array* *has* *changed* *since* *Go* *1.18.*
+*NOTA:* *A* *forma *a* *função* *append* *aumenta* *a* *capacidade* *do* *array*
+*subjacente* *mudou* *desde* *Go* *1.18.*
```
Comment (rschio) [inline _content/tour/por/slices.article:192]:
```suggestion
*NOTA:* *A* *forma* *que* *a* *função* *append* *aumenta* *a* *capacidade* *do* *array*
*subjacente* *mudou* *desde* *Go* *1.18.*
```

File: _content/tour/por/slices.article [hunk 7]
```diff
@@ -436,28 +430,27 @@ add a like to the first user.
...
 
 .image /tour/eng/static/img/sl9.png
 
-You have to be careful to know if a slice is going to be used in an append operation
-during the course of a running program. How you share the slice needs to be
-considered. Sharing individual indexes may not be the best idea. Sharing an
-entire slice value may not work either when appending is in operation. Probably
-using a slice as a field in a struct, and sharing the struct value is a better
-way to go.
+É importante ter cuidado e saber se um `slice` será usadaoem uma operação de anexação durante
```
Comment (rschio) [inline _content/tour/por/slices.article:444]:
```suggestion
É importante ter cuidado e saber se um `slice` será usada em uma operação de anexação durante
```

File: _content/tour/por/slices.article [hunk 8]
```diff
@@ -1,29 +1,31 @@
...
-- *Example* *10:* Three index slicing
+- *Exemplo* *1:* Declarar e Obter o comprimento
+- *Exemplo* *2:* Tipos por Referência
+- *Exemplo* *3:* Anexando slices
+- *Exemplo* *4:* Obter slices de slices
+- *Exemplo* *5:* Slices e Referências
+- *Exemplo* *6:* Strings e slices
+- *Exemplo* *7:* Funções Variádicas
+- *Exemplo* *8:* Range Mecânica do Range
+- *Exemplo* *9:* Traversais Eficientes
```
Comment (rschio) [inline _content/tour/por/slices.article:27]:
```suggestion
- *Exemplo* *9:* Travessias Eficientes
```

File: _content/tour/por/slices.article [hunk 9]
```diff
@@ -248,26 +245,26 @@ Output:
...
 
-** Mutations To The Backing Array
+** Mutação do Array Subjacente
 
-When you use slice2 to change the value of the string at index 0, any slice value
-that is sharing the same backing array (where the address for that index is part
-of that slice’s length) will see the change.
+Quando você usa slice2 para alterar o valor da `string` no índice 0, qualquer valor do `slice`
+que esteja compartilhando o mesmo `array` subjacente (onde o endereço para aquele índice faz
+parte do comprimento dessa fatia) verá a alteração.
```
Comment (rschio) [inline _content/tour/por/slices.article:259]:
```suggestion
parte do comprimento dessa slice) verá a alteração.
```

File: _content/tour/por/slices.article [hunk 10]
```diff
@@ -141,83 +144,77 @@ Output:
...
 
     for record := 1; record <= 102400; record++ {
         data = append(data, fmt.Sprintf("Rec: %d", record))
     }
 
-The append function works with a slice even when the slice is initialized to its
-zero value state. The API design of append is what’s interesting because it uses
-value semantic mutation. Append gets its own copy of a slice value, it mutates its
-own copy, then it returns a copy back to the caller.
+A função `append` funciona com um `slice` mesmo quando o `slice` é inicializad no seu
```
Comment (rschio) [inline _content/tour/por/slices.article:161]:
```suggestion
A função `append` funciona com um `slice` mesmo quando o `slice` é inicializado no seu
```

File: _content/tour/por/slices.article [hunk 11]
```diff
@@ -141,83 +144,77 @@ Output:
...
+A função `inspectSlice` mostra como um `slice` tem um `array` subjacente contíguo com um passo previsível.
+Também mostra como um `slice` tem um comprimento e uma capacidade que podem ser diferentes.
+Observe como a função print itera apenas sobre o comprimento do `slice`.
 
-** Appending With Slices
+** Anexando Slices
 
-The language provides a built-in function called append to add values to an
-existing slice.
+A linguagem fornece uma função built-in chamada append para adicionar valores a un slice existente.
```
Comment (rschio) [inline _content/tour/por/slices.article:153]:
```suggestion
A linguagem fornece uma função built-in chamada append para adicionar valores a um slice existente.
```

**Review by rschio — COMMENTED** [review]
slices comments

File: _content/tour/por/generics-underlying-types.article [hunk 12]
```diff
@@ -100,12 +97,12 @@ construction calls.
...
-possible for the compiler to infer the type. However, in cases where there is
-initialization during construction, the compiler can infer the type.
+Quando se trata de construir esses tipos genéricos até o valor zero, não é possível para o
+compilador inferir o tipo. No entanto, em casos em que há inicialização durante a construção,
+o compilador pode inferir o tipo.
 
-There is an aspect of the spec that focuses on the construction of a generic type to
-its zero value state.
+Existe um aspecto da especificação que se concentra na construção de um tipo genérico
+até seu estado de zero value.
```
[the remark this answers is not in the document]
Comment (luciormoraes) [reply _content/tour/por/generics-underlying-types.article:105]: Indeed. @sergiolpf has suggested: "valor inicial padrão".

**Pull request merged**